File: fixtures/csharp/type_inference/lambda/lambda.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace LambdaApp
{
    public class LambdaDemo
    {
        public static string ApplyTwice(Func<int, int> fn, int value)
        {
            return fn(fn(value)).ToString();
        }

        public static void Main()
        {
            Func<int, string> toLabel = x => $"n={x}";
            var doubled = ApplyTwice(x => x * 2, 21);
            var labels = new List<int> { 1, 2, 3 }.Select(x => x.ToString()).ToList();
            Console.WriteLine($"{toLabel(1)} {doubled} {labels.Count}");
        }
    }
}

File: fixtures/java/type_inference/lambda/lambda.java
import java.util.List;
import java.util.function.Function;
import java.util.stream.Collectors;

public class LambdaDemo {

    public static int applyTwice(Function<Integer, Integer> fn, int value) {
        return fn.apply(fn.apply(value));
    }

    public static void main(String[] args) {
        Function<String, Integer> lengthOf = s -> s.length();
        int doubled = applyTwice(x -> x * 2, 21);
        List<String> labels = List.of(1, 2, 3).stream()
                .map(n -> "n=" + n)
                .collect(Collectors.toList());
        System.out.println(lengthOf.apply("hello") + doubled + labels.size());
    }
}
